huaweicloud/services/aom/resource_huaweicloud_aom_prom_instance.go

```diff
@@ -0,0 +1,214 @@
+package aom
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// API: AOM POST /v1/{project_id}/aom/prometheus
+// API: AOM DELETE /v1/{project_id}/aom/prometheus
+// API: AOM GET /v1/{project_id}/aom/prometheus
+
+const (
+	prometheusInstanceNotExistsCode = "AOM.11017014"
+)
+
+func ResourcePromInstance() *schema.Resource {
```

Comment:
```suggestion
const (
	prometheusInstanceNotExistsCode = "AOM.11017014"
)

// API: AOM POST /v1/{project_id}/aom/prometheus
// API: AOM DELETE /v1/{project_id}/aom/prometheus
// API: AOM GET /v1/{project_id}/aom/prometheus
func ResourcePromInstance() *schema.Resource {
```
It is a recommand to placing API annotations before the main method